cl/corpus_importer/signals.py

```diff
@@ -0,0 +1,126 @@
+from functools import partial
+
+from django.conf import settings
+from django.db import transaction
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from cl.corpus_importer.tasks import make_docket_by_iquery_sweep
+from cl.lib.redis_utils import (
+    acquire_redis_lock,
+    get_redis_interface,
+    make_update_pacer_case_id_key,
+    release_redis_lock,
+)
+from cl.search.models import Court, Docket
+
+
+def update_latest_case_id_and_schedule_iquery_sweep(docket: Docket) -> None:
+    """Updates the latest PACER case ID and schedules iquery retrieval tasks.
+
+    :param docket: The incoming Docket instance.
+    :return: None
+    """
+
+    r = get_redis_interface("CACHE")
+    court_id = str(docket.court_id)
```

Comment:
court_id is a charfield.

Reply:
Yeah, I removed the `str` cast here. I had added it because pycharm was complaining about the court_id being an `int`, but I think it's just something weird from the editor that can't detect it as a character since mypy is not complaining about that.